stock_move_line_auto_fill/models/stock_move.py

```diff
@@ -9,21 +9,28 @@
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    def _get_auto_fill_flag(self):
+        return self.picking_id.auto_fill_operation
+
+    def _get_avoid_lot_assignment_flag(self):
+        return self.picking_id.picking_type_id.avoid_lot_assignment
+
     def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
         """
         Auto-assign as done the quantity proposed for the lots.
         Keep this method to avoid extra write after picking _action_assign
         """
-        self.ensure_one()
-        res = super()._prepare_move_line_vals(quantity, reserved_quant)
-        if not self.picking_id.auto_fill_operation:
-            return res
-        elif self.picking_id.picking_type_id.avoid_lot_assignment and res.get("lot_id"):
-            return res
+        move_line_vals = super()._prepare_move_line_vals(quantity, reserved_quant)
```

Comment:
Let's revert `self.ensure_one()` and the variable as we do not have a strong reason to change them.

```suggestion
        self.ensure_one()
        res = super()._prepare_move_line_vals(quantity, reserved_quant)
```